mllib/src/test/scala/org/apache/spark/ml/regression/RandomForestRegressorSuite.scala

```diff
@@ -105,6 +108,55 @@ class RandomForestRegressorSuite extends SparkFunSuite with MLlibTestSparkContex
       }
   }
 
+  test("Random Forest variance") {
+    val categoricalFeatures = Map.empty[Int, Int]
+    val df: DataFrame = TreeTests.setMetadata(
+      orderedLabeledPoints50_1000, categoricalFeatures, 0)
+
+    // RF with one tree should have the same variance as that of the tree.
+    val rf = new RandomForestRegressor()
+      .setImpurity("variance")
+      .setMaxDepth(30)
+      .setNumTrees(1)
+      .setMaxBins(10)
+      .setFeatureSubsetStrategy("all")
+      .setSubsamplingRate(1.0)
+      .setSeed(123)
+
+    val rfModel = rf.fit(df)
+    val rfVariances = rfModel.transform(df).select("variance").collect()
+
+    val dt = new DecisionTreeRegressor()
+      .setImpurity("variance")
+      .setMaxDepth(30)
+      .setMaxBins(10)
+      .setSeed(123)
+    val dtModel = dt.fit(df)
+    val dtVariances = dtModel.transform(df).select("variance").collect()
+    val nSamples = dtVariances.size
+    (0 to nSamples - 1).foreach { i =>
+      val diff = math.abs(rfVariances(i).getDouble(0) - dtVariances(i).getDouble(0))
```

Comment:
you should be able to use:

`assert(rfVariances(i).getDouble(0) ~== dtVariances(i).getDouble(0) absTol 1e-6)`